virtualmachine/gcp/vm.go

```diff
@@ -56,17 +65,127 @@ type VM struct {
+type Endpoint struct {
+	// Protocol can be one of the following well known protocol strings
+	// (tcp, udp, icmp, esp, ah, sctp) which are supported by GCP
+	Protocol string
+	Ports    []string
+}
+
+// Image represents a GCE image
+type Image struct {
+	Id                uint64 `json:"id,omitempty,string"`
+	CreationTimestamp string `json:"creation_timestamp,omitempty"`
+	// DeprecationStatus indicates whether the image is depcrecated. If the
+	// image is not deprecated it will have an empty string.
+	DeprecationStatus string `json:"deprecation_status,omitempty"`
+	// DeprecationReplacement provides the url of the image which is a
+	// replacement of this deprecated image. If this image is not deprecated
+	// it's value will be an empty string.
+	DeprecationReplacement string `json:"deprecation_replacement,omitempty"`
+	Description            string `json:"description,omitempty"`
+	DiskSizeGb             int64  `json:"disk_size_gb,omitempty,string"`
+	Family                 string `json:"family,omitempty"`
+	Name                   string `json:"name,omitempty"`
+	Status                 string `json:"status,omitempty"`
 }
 
 // Disk represents the GCP Disk.
 // See https://cloud.google.com/compute/docs/disks/?hl=en_US&_ga=1.115106433.702756738.1463769954
 type Disk struct {
-	Name       string
-	DiskType   string
-	DiskSizeGb int
-	AutoDelete bool // Auto delete disk
+	Name        string
+	DiskType    string
+	DiskSizeGb  int
+	AutoDelete  bool // Auto delete disk
+	Description string
+}
+
+// AttachedDisk represents a disk attached to a GCE instance
+type AttachedDisk struct {
+	Name       string `json:"name,omitempty"`
+	DevicePath string `json:"device_path,omitempty"`
+	Boot       *bool  `json:"boot,omitempty"`
+	AutoDelete *bool  `json:"auto_delete,omitempty"`
+	Interface  string `json:"interface,omitempty"`
+}
+
+// Network defines a VPC network in GCP
+type Network struct {
+	Name                  string   `json:"name,omitempty"`
+	Description           string   `json:"description,omitempty"`
+	Id                    uint64   `json:"id,omitempty,string"`
+	AutoCreateSubnetworks *bool    `json:"auto_create_subnetworks,omitempty"`
+	IPv4Range             string   `json:"ipv4_range,omitempty"`
+	CreationTimestamp     string   `json:"creation_timestamp,omitempty"`
+	Subnetworks           []string `json:"subnetworks,omitempty"`
+}
+
+// Subnetwork represents a GCP subnetwork in a region
+type Subnetwork struct {
+	Name              string `json:"name,omitempty"`
+	Description       string `json:"description,omitempty"`
+	Id                uint64 `json:"id,omitempty,string"`
+	CreationTimestamp string `json:"creation_timestamp,omitempty"`
+	GatewayAddress    string `json:"gateway_address,omitempty"`
+	Network           string `json:"network,omitempty"`
+	Region            string `json:"region,omitempty"`
+	IpCidrRange       string `json:"ipv4_range,omitempty"`
+}
+
+// MachinType defines GCP machine type (aka flavors)
+type MachineType struct {
```

Comment:
Should it also include root-disk size?

Reply:
GCP does not have root-disk size defined or limit for each machine type. We do not have any base to provide this information.